src/packageurl/contrib/url2purl.py

```diff
@@ -276,3 +276,23 @@ def build_sourceforge_purl(uri):
 @purl_router.route(cargo_pattern)
 def build_cargo_purl(uri):
     return purl_from_pattern('cargo', cargo_pattern, uri)
+
+
+@purl_router.route('https?://github.com.*/.*')
+def build_github_purl(uri):
+    path = unquote_plus(urlparse(uri).path)
+    segments = [seg for seg in path.split('/') if seg]
+    namespace = segments[0]
+    name = segments[1]
+
+    if len(segments) == 2:
+        return PackageURL('github', name=name, namespace=namespace)
+    
+    if segments[2] == 'tree':
+        version = segments[3]
+
+        if len(segments) > 4:
```

Comment:
This is slightly confusing to read. I think if we should keep the len comparisons on the same indentation level to stay consistent. 

Also, what happens if `len(segments) < 2`? This should be a test case, if it is not already.

Reply:
I have made changes for it :D

Reply:
If they looks good I will squash them :)